jina/flow/__init__.py

```diff
@@ -167,6 +167,7 @@ def __init__(self, args: 'argparse.Namespace' = None, **kwargs):
         super().__init__()
         self.logger = get_logger(self.__class__.__name__)
         self._pod_nodes = OrderedDict()  # type: Dict[str, 'FlowPod']
+        self._pod_needs = list()
```

Comment:
I think this pod_needs do not need to be part of the instance, can't they be computed locally at visualization time?

Reply:
Uhm ok, let me check how to do that